### PR TITLE
fix SettingsImpl to remove hardcoded checkpointRetry

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

Fix the issue of `queue.checkpoint.retry` not taking the config

## What does this PR do?

[SettingsImpl.checkpointRetry](https://github.com/elastic/logstash/blob/v8.4.1/logstash-core/src/main/java/org/logstash/ackedqueue/SettingsImpl.java#L172) is hardcoded to false in builder.
This commit removes the hardcode boolean 'false'

## Why is it important/What is the impact to the user?

Prior to this change, users are unable to set `queue.checkpoint.retry` to true to enable Windows retry on PQ AccessDeniedException in checkpoint

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] add unit test to ensure value is set
- [x] checked the value of `queue.checkpoint.retry` has populated to FileCheckpointIO

## Related issues

- Relates: #12345
- Closes: #14486

### CODE REVIEW

**Comment by prakashguru** [comment]
@kaisecheng will it possible to backport this fix to 7.17 branch? We are in 7.17 and our planned move to 8.x will take some time.

**Comment by kaisecheng** [comment]
@prakashguru It will backport to 7.17

**Review by jsvd — APPROVED** [review]
Nice find :) LGTM

File: logstash-core/src/test/java/org/logstash/ackedqueue/SettingsImplTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                .elementClass(StringElement.class)
+                .build();
+
+        assertEquals(settings.getDirPath(), "PATH/TO/Q");
+        assertEquals(settings.getCapacity(), 10);
+        assertEquals(settings.getMaxUnread(), 1024);
+        assertEquals(settings.getQueueMaxBytes(), 2147483647);
+        assertEquals(settings.getCheckpointMaxAcks(), 1);
+        assertEquals(settings.getCheckpointMaxWrites(), 1);
+        assertTrue(settings.getCheckpointRetry());
```
Comment (andsel) [inline logstash-core/src/test/java/org/logstash/ackedqueue/SettingsImplTest.java:29]: I think that a test named `checkpointRetrySetting` should check just that `checkpointRetry` assumes the expected value.
All the other asserts pollute the intention of this particular test.
Maybe you can split it in 2 tests:
- one to check all the settings
- one to check `checkpointRetry` specifically

or maybe, just change the test name in something more general, like: `verifySettingsGetConfiguredValues`

**Review by andsel — CHANGES_REQUESTED** [review]
I suggested a change in the name of the test or a splitting of the test

File: logstash-core/src/test/java/org/logstash/ackedqueue/SettingsImplTest.java (hunk 1 above)
Comment (kaisecheng) [reply logstash-core/src/test/java/org/logstash/ackedqueue/SettingsImplTest.java:29]: I will change the name. Thanks for catching it.

**Review by andsel — APPROVED** [review]
LGTM

**Comment by kaisecheng** [comment]
@logstashmachine backport 7.17

**Pull request merged**